### PR TITLE
[Magiclysm] New spell: Jar of Force

### PR DESCRIPTION
**Pull request by Light-Wave**
#### Summary
Content "Adds Jar of Force, a magus spell that summons a jar of force that acts as a 3l jar."

#### Purpose of change
Adds a new magus spell that summons a jar made out of force. The jar lasts a very long time (days) and can contain liquids, and be used to boil things in.
#### Describe the solution
Adds the item, the spell, a scroll of the spell, and adds the spell to Wizarding Guide to Backpacking. Adds the scroll to the spell_scroll_tier_1 item group.
The jar is not a great weapon, but is a magic focus, and deals a little bit of pure damage on a hit, since it is made of force.

#### Describe alternatives you've considered
Considered alterations:
- Making the jar permanent at max level.
- Spawning more than one jar at higher levels
- Allowing the jar to be used for crafting (Such as Molotov cocktails and sealed food)
- Making the jar destructible
- Having different balance regarding duration, cost, cast time, melee damage, rarity, and such
- Make it so that adding this scroll to loot tables does not make other scrolls less common
- Make sure the duration does not almost break the spell learning interface due to having so long numbers
- Alter unit that remaining duration is shown in when displaying remaining turns becomes less helpful due to high values

#### Testing
Spawn in a jar and fill it with water. Spawn in another jar and use it for boiling. Boil som water using them. Check that content gets spilled on the ground when timeout happens.

### CODE REVIEW

File: data/mods/Magiclysm/items/ethereal_items.json [hunk 1]
```diff
@@ -744,6 +744,38 @@
       "passive_effects": [ { "has": "WIELD", "condition": "ALWAYS", "values": [ { "value": "ITEM_DAMAGE_COLD", "add": 6 } ] } ]
     }
   },
+  {
+    "id": "jar_3l_force",
+    "type": "GENERIC",
+    "category": "container",
+    "name": { "str": "jar of force" },
```
Comment (Qrox) [inline data/mods/Magiclysm/items/ethereal_items.json:751]:
```suggestion
    "name": { "str": "jar of force", "str_pl": "jars of force" },
```

Comment (Light-Wave) [reply data/mods/Magiclysm/items/ethereal_items.json]: Change is implemented. Thanks for pointing it out.

**Comment by Salty-Panda** [comment]
What in situation when you stop crafting?
Does `in progress item` despawn? Does jar disappear upon completion?

**Comment by Light-Wave** [comment]
> What in situation when you stop crafting?
> Does `in progress item` despawn? Does jar disappear upon completion?

The jar can currently not be used as a crafting component, thus bypassing those issues. In progress liquids are held in hands and not in the jar, so the jar should not affect that; you will just be left without anywhere to put the finished product if the jar despawns while crafting. The jar despawning and removing access to boiling quality tools should work just as well as any other effect that removes a needed tool during crafting.

**Comment by GuardianDll** [comment]
good idea, what about making some changes: make it high level (like 25-35) no class spell (instead of 15 lvl magus)? right now there exist `WITH_CONTAINER` flag in game, but it create simple permanent bottle out of nowhere - maybe it will be better idea to replase it with this spell item in some way?

**Comment by Light-Wave** [comment]
> good idea, what about making some changes: make it high level (like 25-35) no class spell (instead of 15 lvl magus)? Right now there exist `WITH_CONTAINER` flag in game, but it create simple permanent bottle out of nowhere - maybe it will be better idea to replase it with this spell item in some way?

Good suggestions. Making it higher level should be no problem, although I'm not sure someone would want to spend a summer getting really good at summoning jars. But there is no harm in raising the max level.
As for moving it from magus to classless, I'm not entirely sure. My idea was that this jar would be made out of the same stuff that magic missile is, which is the specialty of the magus. But I'm open to having it changed if more people requests it, a stronger case is made, or if @KorGgenT wants it that way.

I like the idea of replacing the `WITH_CONTAINER` with a jar. It would be interesting if the created jar's duration depends on your skill in Jar of Force, instead of the spell used to create the content. A neat piece of spell interaction. This would however require editing c++ code, which I'm not sure that I want to mess around with right now.
It would be neat if you could specify which spell that creates your container in .json, maybe make the flesh pouch watertight and have a spell that creates one filled with acid? I do like the idea of spells interacting with each other like that.

>That said, having both a high-difficulty, no-permanence Classless and a lower-difficulty, permanence-possible Magus spell would make some sense; the former would be useful for alchemy also.

That is an possibility. I'm not entirely sure how KorGgenT balances classless vs. class-based spells.

**Comment by actual-nh** [comment]
> > That said, having both a high-difficulty, no-permanence Classless and a lower-difficulty, permanence-possible Magus spell would make some sense; the former would be useful for alchemy also.
> 
> That is an possibility. I'm not entirely sure how KorGgenT balances classless vs. class-based spells.

See [data/mods/Magiclysm/magic_balance.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/0.F-dev/data/mods/Magiclysm/magic_balance.md) for some info.

**Pull request merged**